### PR TITLE
redesign modal form of search component (fix #466)

### LINKED ISSUE #466: :iphone: Adjust user filters
### How it should look
Filters (icon in the header)

### How it looks like
image

### Figma design
https://www.figma.com/file/pxj9Hhk5AdOxXWJP0dq4gT/Library-App-Spec?node-id=1%3A16

`Note`: add test cases for changed components if it has no corresponding tests

### PR DESCRIPTION
**Pull request by sabrina34352**
image
image

### CODE REVIEW

**Review by mabrur-h — APPROVED** [review]
looks good to me

File: packages/frontapp/src/app/components/RadioButton/index.tsx [hunk 1]
```diff
@@ -100,6 +101,7 @@ interface IRadioGroup {
   name: string;
   options: { name: string; value: string }[];
   onChange?: (value: string) => void;
+  reset?: boolean;
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/RadioButton/index.tsx:104]: if it is boolean, better name it with a question, like `isReset` or 'shouldReset'

File: packages/frontapp/src/app/components/RadioButton/index.tsx [hunk 2]
```diff
@@ -108,13 +110,17 @@ export const RadioGroup: React.FC<IRadioGroup> = ({
   onChange,
   orientation = 'horizontal',
   defaultValue = '',
+  reset,
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/RadioButton/index.tsx:113]: same here, choose different naming

File: packages/frontapp/src/app/components/RadioButton/index.tsx [hunk 3]
```diff
@@ -108,13 +110,17 @@ export const RadioGroup: React.FC<IRadioGroup> = ({
...
 }) => {
   const [checkedValue, setCheckedValue] = useState(defaultValue);
 
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     setCheckedValue(e.target.value);
     onChange && onChange(e.target.value);
   };
+  useEffect(() => {
+    reset && setCheckedValue(defaultValue);
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/RadioButton/index.tsx:122]: same

File: packages/frontapp/src/app/components/RadioButton/index.tsx [hunk 4]
```diff
@@ -108,13 +110,17 @@ export const RadioGroup: React.FC<IRadioGroup> = ({
...
+  }, [reset, defaultValue]);
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/RadioButton/index.tsx:123]: same

File: packages/frontapp/src/app/components/SearchModal/index.tsx [hunk 5]
```diff
@@ -104,64 +117,76 @@ const SearchModal: FC<IProps> = ({
...
                 ))}
             {item.subAttributes.length > numberOfInitialItems && (
-              <SeeMoreButton onClick={() => seeMoreHandler(item.title)}>
+              <SeeMoreButton
+                onClick={() => seeMoreHandler(item.title.toLowerCase())}
+              >
                 {showMore[item.title.toLowerCase() as keyof showMoreStats]
-                  ? 'see less'
-                  : 'see more'}
+                  ? 'See Less'
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/SearchModal/index.tsx:167]: add correspond translation to `resources.ts`

File: packages/frontapp/src/app/components/SearchModal/index.tsx [hunk 6]
```diff
@@ -104,64 +117,76 @@ const SearchModal: FC<IProps> = ({
...
+                  : 'See All'}
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/SearchModal/index.tsx:168]: same, add translation

File: packages/frontapp/src/app/components/SearchModal/index.tsx [hunk 7]
```diff
@@ -104,64 +117,76 @@ const SearchModal: FC<IProps> = ({
...
               </SeeMoreButton>
             )}
           </AttributeWrapper>
         </div>
       ))}
       <ButtonWrapper>
-        <Button value="Apply filters" onClick={() => setApplyFilters(true)} />
+        <Button
+          value="Results"
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/SearchModal/index.tsx:176]: try to add translation for a button value

File: packages/frontapp/src/app/components/SearchModal/index.tsx [hunk 8]
```diff
@@ -104,64 +117,76 @@ const SearchModal: FC<IProps> = ({
...
+            setApplyFilters(true);
+            setReset(false);
+          }}
+        />
         <Button
           type="reset"
           transparent
-          value="Reset all filters"
-          onClick={handleResetClick}
+          value="Reset all"
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/SearchModal/index.tsx:185]: same here, add translation for the button

File: packages/frontapp/src/app/components/UserSearch/index.tsx [hunk 9]
```diff
@@ -38,22 +38,16 @@ const UserSearch: FC<IProps> = ({ setActive }) => {
...
+  const radioBtnHandler = (attributes: Attribute[], value: string) => {
+    if (attributes[0].title === value) {
+      attributes[1].checked = false;
+      attributes[2].checked = false;
+    } else if (attributes[1].title === value) {
+      attributes[0].checked = false;
+      attributes[2].checked = false;
+    } else {
+      attributes[0].checked = false;
+      attributes[1].checked = false;
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/UserSearch/index.tsx:50]: I don't like how this handler looks like. What if we want to add more params to the filter? Check how the radiogroup handler look like on the `Settings` page and rewrite it. Radio button uncheckes other values by default when we are selecting a new one

**Review by Chichmarin17 — CHANGES_REQUESTED** [review]

**Pull request merged**